src/ui/public/courier/fetch/call_client.js

```diff
@@ -34,43 +33,60 @@ export function CallClientProvider(Private, Promise, es) {
   const ABORTED = RequestStatus.ABORTED;
   const DUPLICATE = RequestStatus.DUPLICATE;
 
-  function callClient(requests) {
+  function callClient(searchRequests) {
     // merging docs can change status to DUPLICATE, capture new statuses
-    const statuses = mergeDuplicateRequests(requests);
+    const statuses = mergeDuplicateRequests(searchRequests);
 
     // get the actual list of requests that we will be fetching
-    let requestsToFetch = statuses.filter(isRequest);
-    let execCount = requestsToFetch.length;
+    const requestsToFetch = statuses.filter(isRequest);
+    let requestsToFetchCount = requestsToFetch.length;
 
-    if (!execCount) return Promise.resolve([]);
+    if (requestsToFetchCount === 0) {
+      return Promise.resolve([]);
+    }
 
-    // resolved by respond()
-    let esPromise = undefined;
-    let isRequestAborted = false;
+    // This is how we'll provide the consumer with search responses.
     const defer = Promise.defer();
 
+    const searchStrategiesWithRequests = assignSearchRequestsToSearchStrategies(requestsToFetch);
+
+    // resolved by respondToSearchRequests()
+    const searchRequestPromises = [];
+    let areAllSearchRequestsAborted = false;
+
+    // When we traverse our search requests and send out searches, some of them may fail. We'll
+    // store those that don't fail here.
+    const activeSearchRequests = [];
+
     // for each respond with either the response or ABORTED
-    const respond = function (responses) {
-      responses = responses || [];
-      return Promise.map(requests, function (request, i) {
-        switch (statuses[i]) {
-          case ABORTED:
-            return ABORTED;
-          case DUPLICATE:
-            return request._uniq.resp;
-          default:
-            if (request.aborted) {
-              return ABORTED;
-            }
-
-            const index = _.findIndex(requestsToFetch, request);
-            if (index < 0) {
-              // This means the request failed.
-              return ABORTED;
-            }
-
-            return responses[index];
+    const respondToSearchRequests = (responses = []) => {
+      const aggregatedSearchRequests = searchStrategiesWithRequests.reduce((allSearchRequests, { searchRequests }) => {
```

Comment:
> then I realized it's probably because batching the requests by strategy can cause the requests/responses to be in a different order, so I suppose this is collecting the searchRequest in the order that the responses should be received. Is that right?

Right.

> I think instead of trying to match the ordering of aggregatedResponses we should instead try to pass responses to respondToSearchRequests in the original request order.

Great idea, thanks for the code.